ansible_base/resource_registry/models/resource.py

```diff
@@ -0,0 +1,162 @@
+import uuid
+
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models, transaction
+from django.utils.translation import gettext_lazy as _
+from rest_framework.serializers import ValidationError
+
+from .service_id import service_id
+
+
+def short_service_id():
+    return service_id().split('-')[0]
+
+
+class ResourceType(models.Model):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        from ansible_base.resource_registry.registry import get_registry
+
+        self.resource_registry = get_registry()
+
+    content_type = models.OneToOneField(ContentType, on_delete=models.CASCADE, related_name="resource_type", unique=True)
```

Comment:
Here `content_type` isn't nullable. What about external resources? We need to create a `ResourceType` object to track some type on another server, like say I'm test_app and I need to know about `awx.workflow`. There can be no ContentType for workflow, because it's on the other system.

Reply:
Shared resources should always have a local version of the shared model (ie organizations). I'm not sure that services will need to track resources in other services, outside of maybe the Gateway.

Reply:
From slack: lets investigate caching resource types in the gateway later

Reply:
I want to integrate `ResourceType` specifically into the RBAC app, and this is near-term, at least for doing it in test_app. After giving this a lot of thought, it's just not possible in this form.

Let me make a pitch here - instead of linking `ResourceType` to `contenttypes.ContentType`, what if it _subclassed_ the ContentType model? There are good reasons to let some related models be defined in settings, obviously we're both already working with different user models, but only provided that they share certain common fields/functionality. I really want this to be something that I work with _as if_ it were the ContentType model in some cases. Looking through the rest of this patch, I don't see any hard reasons that we need this to reference the canonical contenttypes app. If you customized the RBAC content type model, you also have to customize the `Permission` model and have that link to your custom content type. Many projects already use a custom permission model, and I think this suits us just fine, because we actually really want a service identifier with the permission model because its the same naming scheme, like an `awx.inventory` should have an `awx.change_inventory` permission.